starter_code/express-spotify-app/app.js

```diff
@@ -0,0 +1,43 @@
+var SpotifyWebApi = require('spotify-web-api-node');
+const express = require('express');
+const expressLayouts = require('express-ejs-layouts');
+const bodyParser = require('body-parser');
+const prettyJson = require('prettyjson');
+const morgan = require('morgan');
+
+const app = express();
+
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+
+app.get('/', (req, res, next) => { 
+  let data = {
+    name: "Spotify",
+    type: "application"
+  };
+  res.render('index', data); // I dont understand why we use response, is it a response
+});                          //  to the response of my request??
+
+
+app.listen(3000,() => {
+  console.log('Listeninng on port 3000');
+});
+
+// Remember to paste here your credentials
+var clientId = 'de05d80b9f2540d4a0d51f5344718aee',
```

Comment:
The code from here until the end has to be before your routes (line 14) so you can use spotifyApi in them.